public/app/plugins/datasource/cloudwatch/components/ConfigEditor.tsx

```diff
@@ -17,16 +19,34 @@ import { store } from 'app/store/store';
 import { CloudWatchDatasource } from '../datasource';
 import { CloudWatchJsonData, CloudWatchSecureJsonData } from '../types';
 
+import { LogGroupSelector } from './LogGroupSelector';
 import { XrayLinkConfig } from './XrayLinkConfig';
 
 export type Props = DataSourcePluginOptionsEditorProps<CloudWatchJsonData, CloudWatchSecureJsonData>;
 
 export const ConfigEditor: FC<Props> = (props: Props) => {
   const { options } = props;
+  const { defaultLogGroups, logsTimeout, defaultRegion } = options.jsonData;
+  const [saved, setSaved] = useState(true);
 
-  const datasource = useDatasource(options.name);
+  const datasource = useDatasource(options.name, saved);
   useAuthenticationWarning(options.jsonData);
-  const logsTimeoutError = useTimoutValidation(props.options.jsonData.logsTimeout);
+  const logsTimeoutError = useTimoutValidation(logsTimeout);
+  useEffect(() => {
+    setSaved(false);
+  }, [options.jsonData.authType, options.secureJsonData, defaultRegion]);
```

Comment:
you are missing some dependencies here, you can see a list in this component: https://github.com/grafana/grafana-aws-sdk-react/blob/main/src/sql/ConfigEditor/ConfigSelect.tsx#L49

I wonder if you could use that ConfigSelect component, you would save some logic and you wouldn't need to implement your own LogGroupSelector (and use the generic ResourceSelector). It may need some changes but it should support multi-selection already. You could also use the ResourceSelector in the QueryEditor, see Athena for an example.

Reply:
Pushed up a change with the correct dependencies while I look into it. I think it would work, but ResourceSelector needs some changes to support the `onInputChange` behavior that the LogGroupSelector has.